china_choropleth_map.Rmd

```diff
@@ -0,0 +1,267 @@
+library(here)
+library(hchinamap)
+library(magrittr)
+library(data.table)
+library(mapproj)
+library(shadowtext)
+library(leafletCN)
+```
+
+## Overview
+From a choropleth map of Covid-19 spread, we could clearly see the number of confirmed cases and severity in each area on the map. We would like to introduce some techniques to plot choropleth map with R. There are different types of maps, like the world map, maps of Continents and maps of Countries etc,. 
+
+Different maps may require different packages to plot. Here we typically explore maps of China. As a running example, we collected China's Covid-19 data and start from **ggplot**, a tool we learned in the lectures. In addition to this, we practiced **hchinamap** and **leafletCN** to plot interactive maps. 
+
+
+## Data Collection
+We web-scarped China's Covid-19 data. The data set includes daily updates of confirmed cases and death for each city in China. 
+```{r}
+
+url = 'https://view.inews.qq.com/g2/getOnsInfo?name=disease_h5&callback=1580373566110'
+x = readLines(url, encoding="UTF-8")
+x = sub("^\\d+", "", x)
+x = sub("^\\(", "", x)
+x = sub("\\)$", "", x)
+y = fromJSON(x)
+d = fromJSON(y$data)
+h = d$areaTree$children[[1]]
+
+names_dic = hash(c("香港", "上海", "新疆", "台湾", "四川", "广东", "陕西",
+                   "福建", "内蒙古", "天津", "河北", "江苏", "山东", "浙江",
+                   "辽宁", "山西", "河南", "重庆", "云南", "北京", "黑龙江", 
+                   "湖南", "广西", "宁夏", "吉林", "西藏", "海南", "澳门",   
+                   "江西", "青海", "湖北", "甘肃", "安徽", "贵州"  ),
+                 c("Hongkong", "Shangahi", "Xinjiang", "Taiwan",  "Sichuan", "Guangdong","Shannxi", 
+                   "Fujian", "Inner Mongolia", "Tianjing", "Hebei","Jiangsu", "Shandong", "Zhejiang",
+                  "Liaoning","Shanxi", "Henan", "Chongqin", "Yunnan", "Beijing","Heilongjiang",
+                   "Hunan", "Guangxi","Ningxia","Jilin","Tibet","Hainan", "Macao", 
+                   "Jiangxi","Qinghai", "Hubei", "Gansu", "Anhui" ,"Guizhou" )
+                 )
+name_en = values(names_dic, keys = h$name, USE.NAMES = FALSE)
+data = data.frame(h$name, name_en, h$total$confirm)
+
+#rename column name
+data<-data %>% 
+  rename(
+    province_CH=h.name,
+    province_EN=name_en,
+    total_confirm=h.total.confirm
+  )
+
+data$province_CH=as.factor(data$province_CH)
+data$province_EN=as.factor(data$province_EN)
+write.csv(file='Heatmap_data.csv',data, fileEncoding =  'UTF-8')
+```
+```{r}
+head(data)
+```
+The dataset we use here is a 34x3 data frame:
+
+province_CH: province name in Chinese
+
+province_EN: province name in English
+
+total_confirm: total confirmed cases 
+
+We are gonna plot total confirmed cases.
+
+## Static Map with ggplot
+Since ggplot does not have build-in China map, we download shape files from *the “Capital of Statistics” webiste* and prepare the shape data for ggplot.
+
+### Prepare the shape data of China
+```{r}
+
+dsn<-paste0(getwd(),"resources/china_choropleth_map/china-province-border-data/bou2_4p.shp") 
```

Comment:
Please remove `getwd()` here

Reply:
Please make the indicated small changes and the chapter will be ready to merge. Thank you!